Add SlimeTech compatibility

PhoenixCodingStuff opened the pull request:
## Short Description
Add SlimeTech compatibility
Change register logic

## Additions/Changes/Removals
Added SlimeTech rubber item
Changed register logic

## Related Issues
N/A

## Checklist
<!-- Here is a little checklist you should follow. -->
<!-- You can click those check boxes after you posted your issue. -->
- [ ] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with base Slimefun and made sure nothing breaks/unexpected happens.
- [ ] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added which may not be obvious to maintainers.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values

Commit:
Update ItemSetup.java

Add SlimeTech compatibility
Change register logic

J3fftw1 commented:
Have you tested this?
im 99% sure this will break recipes in litexpansions that require item id rubber.

WalshyDev commented on src/main/java/dev/j3fftw/litexpansion/ItemSetup.java:
```diff
@@ -43,6 +43,7 @@ final class ItemSetup {
 
     static final ItemSetup INSTANCE = new ItemSetup();
     private final ItemStack glass = new ItemStack(Material.GLASS);
+    private final ItemStack rubberItem;
```
can't be final

WalshyDev commented on src/main/java/dev/j3fftw/litexpansion/ItemSetup.java:
```diff
@@ -87,17 +88,19 @@ private void registerMachines() {
+            this.rubberItem = SlimefunItem.getById("SLIMETECH_RUBBER").getItem();
+        }else if (Bukkit.getServer().getPluginManager().isPluginEnabled("SlimyTreeTaps")) {
+            this.rubberItem = SlimefunItem.getById("RUBBER").getItem();
+        }else {
```
```suggestion
        } else if (Bukkit.getServer().getPluginManager().isPluginEnabled("SlimyTreeTaps")) {
            this.rubberItem = SlimefunItem.getById("RUBBER").getItem();
        } else {
```

WalshyDev commented on src/main/java/dev/j3fftw/litexpansion/ItemSetup.java:
```diff
@@ -87,17 +88,19 @@ private void registerMachines() {
     private void registerRubber() {
-        if (!Bukkit.getServer().getPluginManager().isPluginEnabled("SlimyTreeTaps")) {
-            //Rubber
+        if (Bukkit.getServer().getPluginManager().isPluginEnabled("SlimeTech")) {
```
Why are we prioritising SlimeTech here? SlimyTreeTaps is in 300-400 servers.

J3fftw1 replied:
We shouldnt priortize any the reason this check is here to add support for all rubber if there is any
we should support both imo

ghost commented:
I'll make the necessary code changes in sometime and test it. Also just a clarification does prioritising meaning putting SlimeTech first in the `if` statement? If yes then I'll switch it to SlimyTreeTaps first.

J3fftw1 commented:
> I'll make the necessary code changes in sometime and test it. Also just a clarification does prioritising meaning putting SlimeTech first in the `if` statement? If yes then I'll switch it to SlimyTreeTaps first.

yes :d

Commit:
Update ItemSetup.java

Made rubberItem not final
Added spaces before else
Unprioritized SlimeTech

Give me sometime to test these changes, I will post screenshots after testing

Commit:
Fixed issues

ghost commented:
![Screenshot_2023-08-30-11-12-27-556_net.kdt.pojavlaunch.jpg](https://github.com/Slimefun-Addon-Community/LiteXpansion/assets/97077949/b6348c3c-64ac-4ece-94b6-0d542b05ab2e)

![Screenshot_2023-08-30-11-12-37-039_net.kdt.pojavlaunch.jpg](https://github.com/Slimefun-Addon-Community/LiteXpansion/assets/97077949/6765616a-8989-42fd-b7ad-03781edac1b9)

![Screenshot_2023-08-30-11-15-48-452_net.kdt.pojavlaunch.jpg](https://github.com/Slimefun-Addon-Community/LiteXpansion/assets/97077949/ee82a127-e2c2-4657-9641-e139be75ff27)

![Screenshot_2023-08-30-11-16-07-714_net.kdt.pojavlaunch.jpg](https://github.com/Slimefun-Addon-Community/LiteXpansion/assets/97077949/4276ad8d-944f-4d27-8df7-9dfd079280b8)

J3fftw1 commented on src/main/java/dev/j3fftw/litexpansion/ItemSetup.java:
```diff
@@ -43,6 +43,7 @@ final class ItemSetup {
     private final ItemStack glass = new ItemStack(Material.GLASS);
+    private ItemStack rubberItem;
     private final SlimefunAddon plugin = LiteXpansion.getInstance();
     private boolean initialised;
```
```suggestion
    private final SlimefunAddon plugin = LiteXpansion.getInstance();
    private boolean initialised;
    private ItemStack rubberItem;
```

J3fftw1 commented on src/main/java/dev/j3fftw/litexpansion/ItemSetup.java:
```diff
@@ -87,17 +88,23 @@ private void registerMachines() {
     }
 
     private void registerMiscItems() {
-        final ItemStack rubberItem = SlimefunItem.getById("RUBBER").getItem();
```
please keep the name rubberItem here and rename the other variable.

Commit:
change names & order

ghost commented:
dun

WalshyDev commented on src/main/java/dev/j3fftw/litexpansion/ItemSetup.java:
```diff
@@ -43,9 +43,9 @@ final class ItemSetup {
-    private ItemStack rubberItem;
     private final SlimefunAddon plugin = LiteXpansion.getInstance();
     private boolean initialised;
+    private ItemStack rubber;
```
```suggestion
    private ItemStack rubberItem;
```

WalshyDev commented on src/main/java/dev/j3fftw/litexpansion/ItemSetup.java:
```diff
@@ -90,21 +90,21 @@ private void registerMachines() {
     // Register only if SlimyTreeTaps and SlimeTech do not exist
     private void registerRubber() {
-        ItemStack rubber;
+        ItemStack rubberItem;
```
```suggestion
```

WalshyDev commented on src/main/java/dev/j3fftw/litexpansion/ItemSetup.java:
```diff
@@ -90,21 +90,21 @@ private void registerMachines() {
+        ItemStack rubberItem;
         if (Bukkit.getPluginManager().isPluginEnabled("SlimyTreeTaps")) {
-            rubber = SlimefunItem.getById("RUBBER").getItem();
+            rubberItem = SlimefunItem.getById("RUBBER").getItem();
```
```suggestion
            this.rubberItem = SlimefunItem.getById("RUBBER").getItem();
```

WalshyDev commented on src/main/java/dev/j3fftw/litexpansion/ItemSetup.java:
```diff
@@ -90,21 +90,21 @@ private void registerMachines() {
+            rubberItem = SlimefunItem.getById("RUBBER").getItem();
         } else if (Bukkit.getPluginManager().isPluginEnabled("SlimeTech")) {
-            rubber = SlimefunItem.getById("SLIMETECH_RUBBER").getItem();
+            rubberItem = SlimefunItem.getById("SLIMETECH_RUBBER").getItem();
```
```suggestion
            this.rubberItem = SlimefunItem.getById("SLIMETECH_RUBBER").getItem();
```

WalshyDev commented on src/main/java/dev/j3fftw/litexpansion/ItemSetup.java:
```diff
@@ -90,21 +90,21 @@ private void registerMachines() {
+            rubberItem = SlimefunItem.getById("SLIMETECH_RUBBER").getItem();
         } else {
-            rubber = Items.RUBBER;
+            rubberItem = Items.RUBBER;
```
```suggestion
            this.rubberItem = Items.RUBBER;
```

WalshyDev commented on src/main/java/dev/j3fftw/litexpansion/ItemSetup.java:
```diff
@@ -90,21 +90,21 @@ private void registerMachines() {
             new RubberSynthesizer().register(plugin);
         }
-        this.rubberItem = rubber;
+        this.rubber = rubberItem;
```
```suggestion
```

WalshyDev commented on src/main/java/dev/j3fftw/litexpansion/ItemSetup.java:
```diff
@@ -90,21 +90,21 @@ private void registerMachines() {
 
     private void registerMiscItems() {
-        ItemStack rubber = this.rubberItem;
+        ItemStack rubberItem = this.rubber;
```
```suggestion
```

WalshyDev commented on src/main/java/dev/j3fftw/litexpansion/ItemSetup.java:
```diff
@@ -146,7 +146,7 @@ private void registerMiscItems() {
 
         registerNonPlaceableItem(Items.TIN_CABLE, RecipeType.ENHANCED_CRAFTING_TABLE,
-            rubber, Items.UNINSULATED_TIN_CABLE
+            rubberItem, Items.UNINSULATED_TIN_CABLE
```
```suggestion
            this.rubberItem, Items.UNINSULATED_TIN_CABLE
```

WalshyDev commented on src/main/java/dev/j3fftw/litexpansion/ItemSetup.java:
```diff
@@ -158,7 +158,7 @@ private void registerMiscItems() {
 
         registerNonPlaceableItem(Items.COPPER_CABLE, RecipeType.ENHANCED_CRAFTING_TABLE,
-            rubber, Items.UNINSULATED_COPPER_CABLE
+            rubberItem, Items.UNINSULATED_COPPER_CABLE
```
```suggestion
            this.rubberItem, Items.UNINSULATED_COPPER_CABLE
```

WalshyDev commented on src/main/java/dev/j3fftw/litexpansion/ItemSetup.java:
```diff
@@ -175,7 +175,7 @@ Items.TIN_ITEM_CASING, new ItemStack(Material.REDSTONE), Items.TIN_ITEM_CASING
 
         registerNonPlaceableItem(Items.GOLD_CABLE, RecipeType.ENHANCED_CRAFTING_TABLE,
-            rubber, Items.UNINSULATED_GOLD_CABLE
+            rubberItem, Items.UNINSULATED_GOLD_CABLE
```
```suggestion
            this.rubberItem, Items.UNINSULATED_GOLD_CABLE
```

Commit:
Remove extra "rubber"

ghost commented:
Done, I thought if you do something more than once (`this.rubberItem`) it would be better to create a local variable instead. I fixed it.

J3fftw1 reviewed (changes requested):
rename the global variable to rubber. The diff still is huge.

ghost commented:
Walshy just told me to rename to rubber

J3fftw1 commented:
ugh @WalshyDev we have conflicting reviews
there is no need to make the diff bigger then it is

ghost commented:
Am I supposed to create local variables or nah? Also `rubber` or `rubberItem`

WalshyDev commented:
The diff literally cannot get smaller..

WalshyDev commented:
Changing to rubber would be MORE not less.

J3fftw1 reviewed (approved):
LGTM
